README.md

```diff
@@ -222,6 +222,42 @@ When a pull request is approved and merged into main the deploy script will spin
 
 ![Architecture Diagram](./.images/architecture.svg?raw=true)
 
+```text
+Structure of app - React frontend that renders a form for a user to fill out, Node backend that uses S3 and Dynamo to store and validate forms.
+
+MCPAR - (Managed Care Program Annual Report) report for states to complete annually for each Medicaid managed care program
+
+MLR - (Medical Loss Ratio) report for states to complete annually or when contract terms change
+
+NAAAR - 
+
+Custom JSON & form field creation engine (formFieldFactory) - each report has a custom JSON object, stored in a JSON file, written using a custom schema. This JSON object is referred to as the form template and it is the blueprint from which report form fields are created. It is also used to create routes and navigation elements throughout the app. When provided form fields from this template, the formFieldFactory renders the appropriate form fields. A similar process occurs when a report is exported in PDF preview format. 
+
+Structure Each page has a name, path, and pageType, for example the first page a user sees in the form will be have ‘pageType: standard’ with a ‘verbiage’ object that includes all of the text that precedes the form fields. The the ‘form’ object follows with a unique id and ‘fields’ array that holds one or more objects that represent the individual questions in a form. There are different types of forms as well. If there is a "pageType": "modalDrawer", then instead of a ‘form’ object, it will have a ‘modalForm’ object.
+
+Storage and retrieval of the form template  when a report is created, the form template is stored alongside it in an S3 bucket and reference to that form template’s location is stored in report metadata in Dynamo. This ensures that future changes to the form template do not break existing forms. However, it also means that changes to the form template are generally only forward looking unless an ETL operation is undertaken.
+
+Field IDs Field IDs are immutable, or should at least be treated that way. They should be descriptive of the data captured and should never change so that they can be relied on and referenced by downstream data analysts.
```

Comment:
```suggestion
Field IDs are immutable, or should at least be treated that way. They should be descriptive of the data captured and should never change so that they can be relied on and referenced by downstream data analysts.
```